src/coreclr/vm/method.cpp

```diff
@@ -3021,7 +3021,7 @@ bool MethodDesc::DetermineAndSetIsEligibleForTieredCompilation()
         !IsWrapperStub() &&
 
         // Policy - Generating optimized code is not disabled
-        !IsJitOptimizationDisabledForSpecificMethod())
+        !IsJitOptimizationDisabled())
```

Comment:
Is this change necessary?

This code is explicitly factored to minimize the number of flags checked for each method as the comment above explains. We should not get here when the optimizations are disabled globally. 

`DetermineIsEligibleForTieredCompilationInvariantForAllMethodsInChunk` assert above is trying enforce this invariant among other things.